### DISCUSSION TITLE
[iOS18] Support for dark themed app icon in iOS 18

### DISCUSSION THREAD

**Discussion by hallohallo133** [dotnet/maui#25572] [Ideas]
Apple's iOS 18 supports custom app icons for light, dark and tinted mode, like described here:
(https://developer.apple.com/design/human-interface-guidelines/app-icons#Platform-considerations)

I did some research, but did not find a solution to implement this in MAUI yet. I then thought, if it is maybe possible to directly reference the appicon.appiconset, because in that way Xcode supports it, but I didn't find a solution for this either.

Does any of you know a workaround to implement the custom app icons or do we have to wait for MAUI's official support?

**Comment by imadofficial** [comment]
You can easily fix it by making your own xcassets and putting them in the Platform/iOS/Resources folder. The compiler will pick it up on it's own.

**Reply by coop-patrick** [reply]
> You can easily fix it by making your own xcassets and putting them in the Platform/iOS/Resources folder. The compiler will pick it up on it's own.

Doing this has worked thankyou! For reference I carried out these steps:

1. Created a Platforms/iOS/Resources/Assets.xcassets folder
2. Created a Platforms/iOS/Resources/Assets.xcassets/AppIcon.appiconset folder
3. Created x3 png files within (e.g. 1024-light.png, 1024-dark.png, and 1024-tinted.png), "Build Action" is "Resource".
4. Created a Contents.json file within, referencing the three files

![image](https://github.com/user-attachments/assets/2a8a2d5a-a060-48c3-a710-e3ac7fe9d887)

7. In my Info.plist I've got:

```
<key>XSAppIconAssets</key>
<string>Resources/Assets.xcassets/AppIcon.appiconset</string>
```

9.Contents.json file is:

```
{
  "images": [
    {
      "filename": "1024-light.png",
      "idiom": "universal",
      "platform": "ios",
      "size": "1024x1024"
    },    
    {
      "appearances": [
        {
          "appearance": "luminosity",
          "value": "dark"
        }
      ],
      "filename": "1024-dark.png",
      "idiom": "universal",
      "platform": "ios",
      "size": "1024x1024"
    },
    {
      "appearances": [
        {
          "appearance": "luminosity",
          "value": "tinted"
        }
      ],
      "filename": "1024-tinted.png",
      "idiom": "universal",
      "platform": "ios",
      "size": "1024x1024"
    }
  ]
}
```

10. Seeing as we also use adaptive icons for Android and that's handled in the AndroidManifest, I've removed my existing MauiIcon entry from my csproj. This can be re-entered when Maui implements adaptive icons. This WAS:

`<MauiIcon Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'android'" Include="Resources\AppIcon\appicon.svg" BaseSize="512,512" Color="#FFFFFF" />`
